contracts/DolaFlashMinter.sol

```diff
@@ -0,0 +1,98 @@
+//SPDX-License-Identifier: None
+
+pragma solidity ^0.8.0;
+
+import "./interfaces/IERC3156FlashBorrower.sol";
+import "./interfaces/IERC3156FlashLender.sol";
+import "./utils/Ownable.sol";
+import "./utils/Address.sol";
+import "./ERC20/IERC20.sol";
+import "./ERC20/SafeERC20.sol";
+
+contract DolaFlashMinter is Ownable, IERC3156FlashLender {
+    using SafeERC20 for IERC20;
+    event FlashLoan(address receiver, address token, uint256 value);
+    event FlashLoanRateUpdated(uint256 oldRate, uint256 newRate);
+    event TreasuryUpdated(address oldTreasury, address newTreasury);
+
+    IERC20 public constant dola =
+        IERC20(0x865377367054516e17014CcdED1e7d814EDC9ce4);
+    address public treasury = 0x926dF14a23BE491164dCF93f4c468A50ef659D5B;
+    uint256 public flashMinted;
```

Comment:
This can't ever go higher than `2 ** 112 - 1` right? Seems like it would make sense to just turn it into a `uint112`, and cast to `uint256` where needed?

Edit: Actually, disregard what I said. I ran the benchmark contract below, and it's cheaper to keep it at `uint256`.

```solidity
pragma solidity ^0.8.0;

contract Test {
    uint256 public immutable num;
    uint256 public x;
    uint112 public y;
    
    constructor() {
        num = 1;
        x = 2;
        y = 3;
    }

    // 6459 gas execution cost    
    function setX(uint256 n) public {
        x = x + n;
    }
    
    // 7389 gas execution cost
    function setY(uint256 n) public {
        y = y + uint112(n);
    }
    
    // 1347 gas execution cost
    function getX() public view returns (uint256) {
        return x - num;
    }
    
    // 1368 gas execution cost 
    function getY() public view returns (uint256) {
        return uint256(y) - num;
    }
}

```